### PR TITLE
[py-torch-nvidia-apex] added version 201019 and added dependency on p…

### PR DESCRIPTION
**Pull request by qwertos**
…y-pybind11

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py [hunk 1]
```diff
@@ -14,11 +14,16 @@ class PyTorchNvidiaApex(PythonPackage, CudaPackage):
...
+    # Since the devs don't release their own version numbers,
+    # defining our own. Using the date of the commit referenced
+    # in format YYMMDD, so the number is always increasing.
+    version('201019', commit='8a1ed9e8d35dfad26fb973996319965e4224dcdd')
 
     depends_on('python@3:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
     depends_on('py-torch@0.4:', type=('build', 'run'))
     depends_on('cuda@9:', when='+cuda')
+    depends_on('py-pybind11', when='@201019:,master', type=('build', 'link', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py]:
```suggestion
    depends_on('py-pybind11', when='@201019:', type=('build', 'link', 'run'))
```
`master` is > all numbered releases

File: var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py [hunk 2]
```diff
@@ -14,11 +14,16 @@ class PyTorchNvidiaApex(PythonPackage, CudaPackage):
     phases = ['install']
 
     version('master', branch='master')
+    # Since the devs don't release their own version numbers,
+    # defining our own. Using the date of the commit referenced
+    # in format YYMMDD, so the number is always increasing.
+    version('201019', commit='8a1ed9e8d35dfad26fb973996319965e4224dcdd')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py]: I would call this `2020-10-19`

File: var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py [hunk 3]
```diff
@@ -14,11 +14,13 @@ class PyTorchNvidiaApex(PythonPackage, CudaPackage):
     phases = ['install']
 
     version('master', branch='master')
+    version('2020-10-19', commit='8a1ed9e8d35dfad26fb973996319965e4224dcdd')
 
     depends_on('python@3:', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
     depends_on('py-torch@0.4:', type=('build', 'run'))
     depends_on('cuda@9:', when='+cuda')
+    depends_on('py-pybind11', when='@2020-10-19:', type=('build', 'link', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py]: This is needed for all versions, right? I don't think `when=` is necessary then.

Comment (qwertos) [reply var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py]: I don't know the correct way to notate this situation. Before now, the only version listed was master, and this dependency wasn't listed. I'm going on the assumption, that a previous commit of master didn't have this as a dependency. I'm trying to leave space for a situation where someone needs an older version and py-pybind11 isn't a dependency.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py]: If they need an older version, they can always add it to the package and add a `when` condition. Up to you I guess.

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py (hunk 3 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/py-torch-nvidia-apex/package.py]: eh... I removed it...

**Pull request merged**